sklearn/impute.py

```diff
@@ -42,7 +42,7 @@
 
 def _get_mask(X, value_to_mask):
     """Compute the boolean mask X == missing_values."""
-    if value_to_mask is np.nan:
+    if isinstance(value_to_mask, float) and np.isnan(value_to_mask):
         if X.dtype.kind == "f":
             return np.isnan(X)
```

Comment:
you could use the function `is_scalar_nan` from utils. It's been made for that but I forgot to use it there.
```python
if is_scalar_nan(value_to_mask):
```
instead of
```python
if isinstance(value_to_mask, float) and np.isnan(value_to_mask):
```